keys.js

```diff
@@ -76,5 +77,38 @@ define(function(require) {
 		}
 	}
 
+	/**
+	 * Resolve all key-value pairs in the supplied object and return a promise
+	 * that will always fulfill with the outcome states of all input promises.
+	 * @param {object} object whose key-value pairs will be settled
+	 * @returns {Promise} promise for an object with the mapped and fully
+	 *  settled key-value pairs
+	 */
+	function settle(object) {
+		var keys = Object.keys(object);
+		var results = {};
+
+		if(keys.length === 0) {
+			return toPromise(results);
+		}
+
+		var p = Promise._defer();
+		var resolver = Promise._handler(p);
+		var promises = keys.map(function(k) { return object[k]; });
+
+		when.settle(promises).then(function(states) {
+			populateResults(keys, states, results, resolver);
+		});
+
+		return p;
+	}
+
+		function populateResults(keys, states, results, resolver) {
+			for(var i=0; i<keys.length; i++) {
+				results[keys[i]] = states[i];
+			}
+			resolver.resolve(results);
+		}
```

Comment:
`populateResults` needs to be unindented, to the same level as the `settle` function.  Other than that, :+1: